### PR TITLE
[Refactor] Use tuple as kv cache instead of tensor

### PR DESCRIPTION
**Pull request by lidenghui1110**
### What this PR does / why we need it?

Previously, the graph mode stored KV cache as Tuple while eager mode used Tensor. To enable PD-separated zero-copy mode in later PR:  #950—where both P and D nodes must share the same storage format—this PR unifies the KV cache type to Tuple for both eager and graph modes.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

CI passed.

### CODE REVIEW

File: vllm_ascend/attention/mla_v1.py [hunk 1]
```diff
@@ -1008,13 +1007,21 @@ def _forward_decode(
                 actual_seq_lengths_kv=decode_meta.seq_lens_list,
             )
         else:
+            q = torch.cat([q_nope, q_pe], dim=-1)
+            attn_output = torch.empty(
+                [num_tokens, self.num_heads, self.kv_lora_rank],
+                dtype=q.dtype,
+                device=q.device)
+            k_cache = torch.cat(
```
Comment (MengqingCao) [inline vllm_ascend/attention/mla_v1.py:1017]: I am worried about the extra NPU memory consumption this will bring

File: vllm_ascend/ops/attention.py [hunk 2]
```diff
@@ -154,20 +154,21 @@ def vanilla_chunked_prefill_mla(
...
-        latent_kv_dim + rope_dim)[:, :max_context_len, :]
-    # get kv_c and k_pe
+    cache_kv_c = kv_c_and_k_pe_cache[0].squeeze()
+    cache_k_pe = kv_c_and_k_pe_cache[1].squeeze()
+    
     # cached_kv_c: [batch_size, max_context_len, latent_kv]
     # cached_k_pe: [batch_size, max_context_len, rope_dim]
-    cache_kv_c = cache_kv_c_pe[:, :, :latent_kv_dim]
-    cache_k_pe = cache_kv_c_pe[:, :, latent_kv_dim:]
+    batch_size = query_lens.size(0)
```
Comment (MengqingCao) [inline vllm_ascend/ops/attention.py]: redundant with L154
```suggestion

```

File: vllm_ascend/worker/model_runner_v1.py [hunk 3]
```diff
@@ -2131,35 +2131,39 @@ def initialize_kv_cache(self, kv_cache_config: KVCacheConfig) -> None:
...
+                        num_blocks, block_size, num_kv_heads, head_dim = kv_cache_shape
+                        rope_dim = self.model_config.hf_text_config.qk_rope_head_dim
+                        nope_dim = self.model_config.hf_text_config.kv_lora_rank
+                        assert head_dim == rope_dim + nope_dim, \
+                            f"head_dim({head_dim}) != rope_dim({rope_dim}) + nope_dim({nope_dim})"
+                        nope_cache_shape = (num_blocks, block_size, 
+                                            num_kv_heads, nope_dim)
+                        rope_cache_shape = (num_blocks, block_size, 
+                                            num_kv_heads, rope_dim)
+                        nope_cache = torch.zeros(
```
Comment (MengqingCao) [inline vllm_ascend/worker/model_runner_v1.py]: Let's keep the name unified, all `lantent_kv_cache` or `nope_cache`

**Comment by wtr0504** [comment]
if i choose eager mode ,does this change matters ?

File: vllm_ascend/attention/mla_v1.py (hunk 1 above)
Comment (lidenghui1110) [reply vllm_ascend/attention/mla_v1.py:1017]: Yes. It will use extra NPU memory in torch.cat, do you have any better suggestion? 
I noticed that in prefill, we alse use cat to do such things, like: [line:830](https://github.com/vllm-project/vllm-ascend/blob/70588531ab8b36fc04c0d067b5fd8d3ff20c3460/vllm_ascend/attention/mla_v1.py#L830), so the consumption should have been calcaulated in warmup stage.
And in deepseek-r1, the consumption should be per layer kv cache size, roughly about 15G/61 = 251M, maybe it is affordable.

File: vllm_ascend/ops/attention.py (hunk 2 above)
Comment (lidenghui1110) [reply vllm_ascend/ops/attention.py]: done

File: vllm_ascend/worker/model_runner_v1.py (hunk 3 above)
Comment (lidenghui1110) [reply vllm_ascend/worker/model_runner_v1.py]: yes, modified to use kv_cache_nope and kv_cache_pe here. And in mla_v1.py, use kv_c and k_pe for short as before

**Comment by lidenghui1110** [comment]
> if i choose eager mode ,does this change matters ?

no, it won't have any influence on any mode.

**Comment by wtr0504** [comment]
> > if i choose eager mode ,does this change matters ?
> 
> no, it won't have any influence on any mode.
thank a lot

File: vllm_ascend/attention/mla_v1.py (hunk 1 above)
Comment (MengqingCao) [reply vllm_ascend/attention/mla_v1.py:1017]: Acctualy I have no good suggestions on it, either. We seems must do this concat. But I think we can remove this until the ring attention could be enabled, right? If so, I think this change is acceptable. also cc @ganyi1996ppo

Comment (ganyi1996ppo) [reply vllm_ascend/attention/mla_v1.py:1017]: This line can be wipe out if ringmla is public accessible, and seems most of the change in this PR  already contains in the PR https://github.com/vllm-project/vllm-ascend/pull/950 , can you refactor it after that PR merge? Or there will be lots of conflict, which may bring more barrier for the 950 to merge....... cc @wangxiyuan @Yikun

Comment (lidenghui1110) [reply vllm_ascend/attention/mla_v1.py:1017]: @ganyi1996ppo It's OK. If all the changes in this PR has been done in [#950](https://github.com/vllm-project/vllm-ascend/pull/950), this PR can be closed. If so, hope it can be merged asap, other works rely on kv cache will rely on it.
